mirrord/layer/src/file/ops.rs

```diff
@@ -217,6 +219,80 @@ pub(crate) fn fdopen(fd: RawFd, rawish_mode: Option<&CStr>) -> Detour<*mut FILE>
     Detour::Success(result)
 }
 
+/// creates a directory stream for the `remote_fd` in the agent
+#[tracing::instrument(level = "trace")]
+pub(crate) fn fdopendir(fd: RawFd) -> Detour<usize> {
+    // usize == ptr size
+    // we don't return a pointer to an address that contains DIR
+
+    let open_files = OPEN_FILES.lock()?;
+    let remote_fd = open_files.get(&fd).ok_or(Bypass::LocalFdNotFound(fd))?;
+
+    let (dir_channel_tx, dir_channel_rx) = oneshot::channel();
+
+    let open_dir_request = OpenDir {
+        remote_fd: *remote_fd,
+        dir_channel_tx,
+    };
+
+    blocking_send_file_message(HookMessageFile::OpenDir(open_dir_request))?;
+
+    let _ = dir_channel_rx.blocking_recv()??;
+
+    let fake_local_dir_name = CString::new(remote_fd.to_string())?;
+    let local_dir_fd = unsafe { create_local_fake_file(fake_local_dir_name, *remote_fd) }?;
```

Comment:
That's the fd of the open, not of the stream being reused instead of letting agent generate another one..

Reply:
Ah i see what you mean, ill try and fix it. thanks for clarifying